contracts/PinningManager.sol

```diff
@@ -195,22 +194,20 @@ contract PinningManager {
             isPastCurrentEndTime,
             "PinningManager: Request already active"
         );
-
         // If request exist from past, lets have clean state. Eq. force withdraw of previous money.
         if(isPastCurrentEndTime) {
             require(offer.capacity != 0, "PinningManager: provider discontinued service");
             //NO_OVERFLOW reasoning: numberOfPeriodsDeposited always bigger or equal to numberOfPeriodsWithdrawn
             uint256 toTransfer = (request.numberOfPeriodsDeposited - request.numberOfPeriodsWithdrawn).mul(request.chosenPrice);
             request.numberOfPeriodsWithdrawn = 0;
-            request.startDate = 0;
             provider.transfer(toTransfer); //TODO: transfer is not best practice: https://diligence.consensys.net/blog/2019/09/stop-using-soliditys-transfer-now/
             emit EarningsWithdrawn(requestReference);
         } else {
             request.size = size;
             offer.capacity = uint128(offer.capacity.sub(size));
         }
         //NO_OVERFLOW reasoning: chosenPrice is verified to not be zero in this function call
-        uint256 numberOfPeriodsDeposited = msg.value / chosenPrice;
+        uint256 numberOfPeriodsDeposited = (msg.value / size) / chosenPrice;
```

Comment:
I would argue to use `request.size` instead of `size`. I am not completely sure, but I believe there could be attack vector, where you create Request with size A you let it expire and then you "renew" the Request but you call the `newRequest` with different size B that would be smaller than size A and you would achieve more periods deposited for the price of size A.

Reply:
I fixed it with `request.size` which should not have any effects on the working, but might prevent this attack.